server/services/v1/workers/worker.go

```diff
@@ -101,10 +102,22 @@ func (w *Worker) jitterSleep() {
 }
 
 func (w *Worker) Start() {
+	defer catchWorkerPanics()
 	w.jitterSleep()
 	w.Loop()
 }
 
+func catchWorkerPanics() {
+	if r := recover(); r != nil {
+		stackTrace := string(debug.Stack())
+		if err, ok := r.(error); ok {
+			log.Err(err).Str("stacktrace", stackTrace).Msg("worker panic")
+		} else {
+			log.Error().Str("stacktrack", stackTrace).Msgf("worker panic %v", err)
```

Comment:
Send the error as a separate argument with Err(), so it's searchable, rather than putting it in the message as text.

Reply:
@pboros I do that above if the value from recover can be converted into an error. Otherwise we have to put it in the msg

Reply:
Ah ok, makes sense.